tests/testthat/test_agg.R

```diff
@@ -0,0 +1,346 @@
+value_cols <- c("value")
+
+# set up expected output data.table
+# expected value is number of single year age groups in aggregate interval
+expected_dt_female <- CJ(year_start = 2005:2009, sex = "female")
+gen_end(
+  expected_dt_female,
+  id_cols = c("year_start", "sex"),
+  col_stem = "year",
+  right_most_endpoint = 2010
+)
+expected_dt_female <- expected_dt_female[
+  , data.table(age_mapping),
+  by = c("year_start", "year_end", "sex")
+]
+expected_dt_female[, value := rep(c(96, 5, 45, 30, 11), 5)]
+
+expected_dt_male <- CJ(year_start = 2005, year_end = 2010, sex = "male")
+expected_dt_male <- expected_dt_male[
+  , data.table(age_mapping),
+  by = c("year_start", "year_end", "sex")
+]
+expected_dt_male[, value := c(96, 5, 45, 30, 11) * 5]
+
+expected_dt <- rbind(expected_dt_female, expected_dt_male, use.names = T)
+setcolorder(expected_dt, c(id_cols, value_cols))
+setkeyv(expected_dt, id_cols)
+
+description <- "aggregating an interval variable with varying interval id cols
+works"
+testthat::test_that(description, {
+  output_dt <- agg(
+    dt = input_dt,
+    id_cols = id_cols, value_cols = value_cols,
+    col_stem = "age", col_type = "interval",
+    mapping = age_mapping,
+    present_agg_severity = "skip"
+  )
+  testthat::expect_identical(output_dt, expected_dt)
+})
+
+description <- "throws error when aggregating an interval variable and there
+are missing intervals"
+testthat::test_that(description, {
+
+  new_input_dt <- input_dt[!(sex == "female" & (age_start <= 2 | age_end >= 95))]
+  setkeyv(new_input_dt, id_cols)
+
+  new_expected_dt <- expected_dt[!(sex == "female" & (age_start <= 2 | age_end >= 95))]
+  setkeyv(new_expected_dt, id_cols)
+
+  # check severity
+  expect_error(
+    agg(dt = new_input_dt,
+        id_cols = id_cols, value_cols = value_cols,
+        col_stem = "age", col_type = "interval",
+        mapping = age_mapping,
+        missing_dt_severity = "stop",
+        present_agg_severity = "skip"
+    ),
+    regexp = "missing making it impossible to collapse"
+  )
+
+  output_dt <- agg(
+    dt = new_input_dt,
+    id_cols = id_cols, value_cols = value_cols,
+    col_stem = "age", col_type = "interval",
+    mapping = age_mapping,
+    missing_dt_severity = "none",
+    present_agg_severity = "skip"
+  )
+  expect_identical(output_dt, new_expected_dt)
+})
+
+description <- "error is thrown when interval aggregate is already included
+in input or works when `present_agg_severity = 'none'`"
+testthat::test_that(description, {
+
+  new_input_dt <- unique(rbind(input_dt, expected_dt))
+  setkeyv(new_input_dt, id_cols)
+
+  setkeyv(expected_dt, id_cols)
+
+  expect_error(
+    agg(dt = new_input_dt,
+        id_cols = id_cols, value_cols = value_cols,
+        col_stem = "age", col_type = "interval",
+        mapping = age_mapping
+    ),
+    regexp = "overlapping intervals were identified in `dt`"
+  )
+})
+
+
+# Aggregate categorical variable with multiple levels in mapping ----------
+
+# Inputs:
```

Comment:
Maybe should just use this instead of the `sex` aggregation example above for tests related to categorical aggregation?

Reply:
Unless these tests are running very slowly, I actually think it's a good thing to split out the multi-level testing from the basic tests you've set up for sex aggregation. That way it might be easier to isolate different issues.

Reply:
Sounds good, left these in